### ISSUE TITLE
[Internal] Make our gradle build consistent

### ISSUE THREAD

**Issue by samtstern** [firebase/FirebaseUI-Android#1462]
I have been fighting the build all day so I just want to log some of the issues here so that I can try and fix them.

This is what we run for our builds;
`./gradlew clean assembleDebug proguard-tests:build check`

It should always give the same result, on any machine.  However I have run it about 100x today and run into sporadic issues like:

  * `Cannot invoke method containsKey() on null object` (from various tasks)
  * `Task :lint:compileKotlin FAILED`
  * `Task :common:generateDebugRFile FAILED`
  * `Task :library:packageDebugResources FAILED`

I'd say of the 100 builds I tried, maybe 25 succeded :-(

**Comment by samtstern** [comment]
Also might be a good opportunity to get ready for Gradle 5.0

**Comment by SUPERCILEX** [comment]
Ouch, yeah I haven't seen those kinds of issues 😞.

**Comment by samtstern** [comment]
@SUPERCILEX do you know of any way to make importing the project easier?  Could we have like a root `build.gradle` that just references a `kts` file with all the meat?

**Comment by samtstern** [comment]
Yeah maybe just waiting will be the thing to do.  I'm surprised that more of the Android community hasn't jumped to Kotlin scripts.  

Also still can't believe you don't use AS!

**Comment by SUPERCILEX** [comment]
Yeah, I dunno.

I do use AS Canary for Android specific stuff, but IntelliJ has the latest and greatest in terms of pure coding tools. 😉
